### PR TITLE
Describe Arquillian test architecture and configuration options in TESTS.md AGPUSH-1435

### PR DESCRIPTION
**Pull request by lfryc**
https://issues.jboss.org/browse/AGPUSH-1435

### CODE REVIEW

File: TESTS.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+Since tests are deployed to running container, they can reuse anything that is deployed with them:
+
+* inject CDI (`@Inject`) or EJB (`@EJB`) beans
+* inject resources such as data sources or JMS destinations (`@Resource`)
+* mock dependencies using Mockito (see `UnifiedPushArchive#withMockito`)
+
+Tests can leverage basic JUnit features such as `@Before` and `@After`, `@Test(timeout)`, `@Test(expected)` or even `@Rule`, etc.
+
+Tests for cluster
```
Comment (matzew) [inline TESTS.md]: @lfryc should some info about multicast on OSX in here - or a link to https://github.com/aerogear/aerogear-unifiedpush-server#getting-started-with-clustered-servers ?

**Comment by matzew** [comment]
@lfryc outside of the testing configuration - should we have some infos on recommended configuration for production, e.g. in the guide ?

**Comment by lfryc** [comment]
@matzew I believe that the current configuration follows sensible default for production, but let me think about it...

File: TESTS.md (hunk 1 above)
Comment (lfryc) [inline TESTS.md]: :+1: will add

**Comment by matzew** [comment]
@lfryc I think so too :-) 

Things will evolve over time, like the iOS batchSize issue, brought up in the community

**Comment by lfryc** [comment]
> Things will evolve over time, ... as brought up in the community

:100:

**Comment by matzew** [comment]
landed on MASTER!!!!!

**Pull request closed without merging**